okhttp-ws/src/main/java/com/squareup/okhttp/internal/ws/RealWebSocket.java

```diff
@@ -45,9 +46,10 @@
   /** True after calling {@link #close(int, String)}. No writes are allowed afterward. */
   private volatile boolean writerSentClose;
   /** True after a close frame was read by the reader. No frames will follow it. */
-  private volatile boolean readerSentClose;
-  /** Lock required to negotiate closing the connection. */
-  private final Object closeLock = new Object();
+  private boolean readerSentClose;
```

Comment:
This no longer requires locking or even `volatile`. It's value is only read and written on the reader thread.